Lib es2023 not supported

tsconfig.json
```
"lib": [
      "dom",
      "dom.iterable",
      "es2023",
      "esnext"
    ]
```

```
❯ scip-typescript index --progress-bar
node_modules/@tsconfig/node18/tsconfig.json(8,13): error TS6046: Argument for '--lib' option must be: 'es5', 'es6', 'es2015', 'es7', 'es2016', 'es2017', 'es2018', 'es2019', 'es2020', 'es2021', 'es2022', 'esnext', 'dom', 'dom.iterable', 'webworker', 'webworker.importscripts', 'webworker.iterable', 'scripthost', 'es2015.core', 'es2015.collection', 'es2015.generator', 'es2015.iterable', 'es2015.promise', 'es2015.proxy', 'es2015.reflect', 'es2015.symbol', 'es2015.symbol.wellknown', 'es2016.array.include', 'es2017.object', 'es2017.sharedmemory', 'es2017.string', 'es2017.intl', 'es2017.typedarrays', 'es2018.asyncgenerator', 'es2018.asynciterable', 'es2018.intl', 'es2018.promise', 'es2018.regexp', 'es2019.array', 'es2019.object', 'es2019.string', 'es2019.symbol', 'es2019.intl', 'es2020.bigint', 'es2020.date', 'es2020.promise', 'es2020.sharedmemory', 'es2020.string', 'es2020.symbol.wellknown', 'es2020.intl', 'es2020.number', 'es2021.promise', 'es2021.string', 'es2021.weakref', 'es2021.intl', 'es2022.array', 'es2022.error', 'es2022.intl', 'es2022.object', 'es2022.sharedmemory', 'es2022.string', 'esnext.array', 'esnext.symbol', 'esnext.asynciterable', 'esnext.intl', 'esnext.bigint', 'esnext.string', 'esnext.promise', 'esnext.weakref'.
```

--- a/package.json
+++ b/package.json
@@ -37,7 +37,7 @@
     "google-protobuf": "^3.21.2",
     "pretty-ms": "^7.0.1",
     "progress": "^2.0.3",
-    "typescript": "^4.8.4"
+    "typescript": "^5.3.3"
   },
   "devDependencies": {
     "@sourcegraph/eslint-config": "0.37.1",

--- a/snapshots/output/pure-js/src/main.js
+++ b/snapshots/output/pure-js/src/main.js
@@ -22,11 +22,11 @@
 //                   ^ definition pure-js 1.0.0 src/`main.js`/print_fib().(a)
 //                   documentation ```ts\n(parameter) a: any\n```
     console.log(fib(a))
-//  ^^^^^^^ reference typescript 4.9.5 lib/`lib.dom.d.ts`/console.
+//  ^^^^^^^ reference typescript 5.3.3 lib/`lib.dom.d.ts`/console.
 //  ^^^^^^^ reference @types/node 20.10.5 `globals.d.ts`/global/console.
 //  ^^^^^^^ reference @types/node 20.10.5 `console.d.ts`/`"node:console"`/global/console/
 //  ^^^^^^^ reference @types/node 20.10.5 `console.d.ts`/`"node:console"`/global/console.
-//          ^^^ reference typescript 4.9.5 lib/`lib.dom.d.ts`/Console#log().
+//          ^^^ reference typescript 5.3.3 lib/`lib.dom.d.ts`/Console#log().
 //          ^^^ reference @types/node 20.10.5 `console.d.ts`/`"node:console"`/global/Console#log().
 //              ^^^ reference pure-js 1.0.0 src/`main.js`/fib().
 //                  ^ reference pure-js 1.0.0 src/`main.js`/print_fib().(a)
@@ -71,7 +71,7 @@
   
   function forever() {
 //         ^^^^^^^ definition pure-js 1.0.0 src/`main.js`/forever().
-//         documentation ```ts\nfunction forever(): any\n```
+//         documentation ```ts\nfunction forever(): never\n```
     return forever()
 //         ^^^^^^^ reference pure-js 1.0.0 src/`main.js`/forever().
   }

--- a/snapshots/output/react/src/LoaderInput.tsx
+++ b/snapshots/output/react/src/LoaderInput.tsx
@@ -1,8 +1,8 @@
   import React from 'react'
 // definition react-example 1.0.0 src/`LoaderInput.tsx`/
 //documentation ```ts\nmodule "LoaderInput.tsx"\n```
-//       ^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/React/
-//                  ^^^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/
+//       ^^^^^ reference @types/react 18.2.39 `index.d.ts`/React/
+//                  ^^^^^^^ reference @types/react 18.2.39 `index.d.ts`/
   
   /** Takes loading prop, input component as child */
   interface Props {
@@ -15,15 +15,15 @@
     children: React.ReactNode
 //  ^^^^^^^^ definition react-example 1.0.0 src/`LoaderInput.tsx`/Props#children.
 //  documentation ```ts\n(property) children: ReactNode\n```
-//            ^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/React/
-//                  ^^^^^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/React/ReactNode#
+//            ^^^^^ reference @types/react 18.2.39 `index.d.ts`/React/
+//                  ^^^^^^^^^ reference @types/react 18.2.39 `index.d.ts`/React/ReactNode#
   }
   
   export const LoaderInput: React.FunctionComponent<Props> = ({
 //             ^^^^^^^^^^^ definition react-example 1.0.0 src/`LoaderInput.tsx`/LoaderInput.
 //             documentation ```ts\nvar LoaderInput: FunctionComponent<Props>\n```
-//                          ^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/React/
-//                                ^^^^^^^^^^^^^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/React/FunctionComponent#
+//                          ^^^^^ reference @types/react 18.2.39 `index.d.ts`/React/
+//                                ^^^^^^^^^^^^^^^^^ reference @types/react 18.2.39 `index.d.ts`/React/FunctionComponent#
 //                                                  ^^^^^ reference react-example 1.0.0 src/`LoaderInput.tsx`/Props#
     loading,
 //  ^^^^^^^ definition local 3
@@ -35,23 +35,23 @@
 //  ^^^^^^^^ reference react-example 1.0.0 src/`LoaderInput.tsx`/Props#children.
   }) => (
     <div className="hello">
-//   ^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/global/JSX/IntrinsicElements#div.
-//       ^^^^^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/React/HTMLAttributes#className.
+//   ^^^ reference @types/react 18.2.39 `index.d.ts`/global/JSX/IntrinsicElements#div.
+//       ^^^^^^^^^ reference @types/react 18.2.39 `index.d.ts`/React/HTMLAttributes#className.
       {children}
 //     ^^^^^^^^ reference local 4
       {loading && <p>spinner</p>}
 //     ^^^^^^^ reference local 3
-//                 ^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/global/JSX/IntrinsicElements#p.
-//                            ^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/global/JSX/IntrinsicElements#p.
+//                 ^ reference @types/react 18.2.39 `index.d.ts`/global/JSX/IntrinsicElements#p.
+//                            ^ reference @types/react 18.2.39 `index.d.ts`/global/JSX/IntrinsicElements#p.
     </div>
-//    ^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/global/JSX/IntrinsicElements#div.
+//    ^^^ reference @types/react 18.2.39 `index.d.ts`/global/JSX/IntrinsicElements#div.
   )
   
   export const LoaderInput2: React.FunctionComponent<Props> = props => {
 //             ^^^^^^^^^^^^ definition react-example 1.0.0 src/`LoaderInput.tsx`/LoaderInput2.
 //             documentation ```ts\nvar LoaderInput2: FunctionComponent<Props>\n```
-//                           ^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/React/
-//                                 ^^^^^^^^^^^^^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/React/FunctionComponent#
+//                           ^^^^^ reference @types/react 18.2.39 `index.d.ts`/React/
+//                                 ^^^^^^^^^^^^^^^^^ reference @types/react 18.2.39 `index.d.ts`/React/FunctionComponent#
 //                                                   ^^^^^ reference react-example 1.0.0 src/`LoaderInput.tsx`/Props#
 //                                                            ^^^^^ definition local 6
 //                                                            documentation ```ts\n(parameter) props: Props\n```

--- a/snapshots/output/react/src/MyTSXElement.tsx
+++ b/snapshots/output/react/src/MyTSXElement.tsx
@@ -1,8 +1,8 @@
   import React from 'react'
 // definition react-example 1.0.0 src/`MyTSXElement.tsx`/
 //documentation ```ts\nmodule "MyTSXElement.tsx"\n```
-//       ^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/React/
-//                  ^^^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/
+//       ^^^^^ reference @types/react 18.2.39 `index.d.ts`/React/
+//                  ^^^^^^^ reference @types/react 18.2.39 `index.d.ts`/
   
   export interface MyProps {}
 //                 ^^^^^^^ definition react-example 1.0.0 src/`MyTSXElement.tsx`/MyProps#
@@ -11,8 +11,8 @@
   export const MyTSXElement: React.FunctionComponent<MyProps> = ({}) => (<p></p>)
 //             ^^^^^^^^^^^^ definition react-example 1.0.0 src/`MyTSXElement.tsx`/MyTSXElement.
 //             documentation ```ts\nvar MyTSXElement: FunctionComponent<MyProps>\n```
-//                           ^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/React/
-//                                 ^^^^^^^^^^^^^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/React/FunctionComponent#
+//                           ^^^^^ reference @types/react 18.2.39 `index.d.ts`/React/
+//                                 ^^^^^^^^^^^^^^^^^ reference @types/react 18.2.39 `index.d.ts`/React/FunctionComponent#
 //                                                   ^^^^^^^ reference react-example 1.0.0 src/`MyTSXElement.tsx`/MyProps#
-//                                                                        ^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/global/JSX/IntrinsicElements#p.
-//                                                                            ^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/global/JSX/IntrinsicElements#p.
+//                                                                        ^ reference @types/react 18.2.39 `index.d.ts`/global/JSX/IntrinsicElements#p.
+//                                                                            ^ reference @types/react 18.2.39 `index.d.ts`/global/JSX/IntrinsicElements#p.

--- a/snapshots/output/react/src/UseMyTSXElement.tsx
+++ b/snapshots/output/react/src/UseMyTSXElement.tsx
@@ -1,8 +1,8 @@
   import React from "react";
 // definition react-example 1.0.0 src/`UseMyTSXElement.tsx`/
 //documentation ```ts\nmodule "UseMyTSXElement.tsx"\n```
-//       ^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/React/
-//                  ^^^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/
+//       ^^^^^ reference @types/react 18.2.39 `index.d.ts`/React/
+//                  ^^^^^^^ reference @types/react 18.2.39 `index.d.ts`/
   
   import { MyProps, MyTSXElement } from "./MyTSXElement";
 //         ^^^^^^^ reference react-example 1.0.0 src/`MyTSXElement.tsx`/MyProps#
@@ -12,8 +12,8 @@
   export const _: React.FunctionComponent<MyProps> =
 //             ^ definition react-example 1.0.0 src/`UseMyTSXElement.tsx`/_.
 //             documentation ```ts\nvar _: FunctionComponent<MyProps>\n```
-//                ^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/React/
-//                      ^^^^^^^^^^^^^^^^^ reference @types/react 18.2.39 `ts5.0`/`index.d.ts`/React/FunctionComponent#
+//                ^^^^^ reference @types/react 18.2.39 `index.d.ts`/React/
+//                      ^^^^^^^^^^^^^^^^^ reference @types/react 18.2.39 `index.d.ts`/React/FunctionComponent#
 //                                        ^^^^^^^ reference react-example 1.0.0 src/`MyTSXElement.tsx`/MyProps#
       ({}) => (<MyTSXElement></MyTSXElement>)
 //              ^^^^^^^^^^^^ reference react-example 1.0.0 src/`MyTSXElement.tsx`/MyTSXElement.

--- a/snapshots/output/syntax/src/accessors.ts
+++ b/snapshots/output/syntax/src/accessors.ts
@@ -138,6 +138,6 @@
     g(D.length)
 //  ^ reference syntax 1.0.0 src/`accessors.ts`/g().
 //    ^ reference syntax 1.0.0 src/`accessors.ts`/D#
-//      ^^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Function#length.
+//      ^^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Function#length.
   }
   

--- a/snapshots/output/syntax/src/destructuring.ts
+++ b/snapshots/output/syntax/src/destructuring.ts
@@ -25,7 +25,7 @@
 //                   ^^^^^ reference syntax 1.0.0 src/`destructuring.ts`/props.
     return [props].map(({ a }) => a + b)
 //          ^^^^^ reference syntax 1.0.0 src/`destructuring.ts`/props.
-//                 ^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Array#map().
+//                 ^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Array#map().
 //                        ^ definition local 10
 //                        documentation ```ts\n(parameter) a: number\n```
 //                        ^ reference syntax 1.0.0 src/`destructuring.ts`/Props#a.
@@ -42,7 +42,7 @@
 //               ^^^^^ reference syntax 1.0.0 src/`destructuring.ts`/props.
     return [[b]].map(([a]) => a.a)
 //           ^ reference local 15
-//               ^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Array#map().
+//               ^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Array#map().
 //                     ^ definition local 21
 //                     documentation ```ts\n(parameter) a: Props\n```
 //                            ^ reference local 21
@@ -58,7 +58,7 @@
 //                  ^^^^^ reference syntax 1.0.0 src/`destructuring.ts`/props.
     return [[props]].map(([{ a }]) => a + b.a)
 //           ^^^^^ reference syntax 1.0.0 src/`destructuring.ts`/props.
-//                   ^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Array#map().
+//                   ^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Array#map().
 //                           ^ definition local 36
 //                           documentation ```ts\n(parameter) a: number\n```
 //                           ^ reference syntax 1.0.0 src/`destructuring.ts`/Props#a.

--- a/snapshots/output/syntax/src/import.ts
+++ b/snapshots/output/syntax/src/import.ts
@@ -43,14 +43,14 @@
   export function dynamicImport(): Promise<void> {
 //                ^^^^^^^^^^^^^ definition syntax 1.0.0 src/`import.ts`/dynamicImport().
 //                documentation ```ts\nfunction dynamicImport(): Promise<void>\n```
-//                                 ^^^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Promise#
-//                                 ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.iterable.d.ts`/Promise#
-//                                 ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.promise.d.ts`/Promise.
-//                                 ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.symbol.wellknown.d.ts`/Promise#
-//                                 ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2018.promise.d.ts`/Promise#
+//                                 ^^^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Promise#
+//                                 ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.iterable.d.ts`/Promise#
+//                                 ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.promise.d.ts`/Promise.
+//                                 ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.symbol.wellknown.d.ts`/Promise#
+//                                 ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2018.promise.d.ts`/Promise#
     return import('./function').then(c => c.newFunction())
 //                ^^^^^^^^^^^^ reference syntax 1.0.0 src/`function.ts`/
-//                              ^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Promise#then().
+//                              ^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Promise#then().
 //                                   ^ definition local 3
 //                                   documentation ```ts\n(parameter) c: typeof import("/src/function")\n```
 //                                        ^ reference local 3

--- a/snapshots/output/syntax/src/infer-relationship.ts
+++ b/snapshots/output/syntax/src/infer-relationship.ts
@@ -12,11 +12,11 @@
 //         ^^^^^^ definition syntax 1.0.0 src/`infer-relationship.ts`/random().
 //         documentation ```ts\nfunction random(): number\n```
     return Math.random()
-//         ^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Math#
-//         ^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Math.
-//         ^^^^ reference typescript 4.9.5 lib/`lib.es2015.core.d.ts`/Math#
-//         ^^^^ reference typescript 4.9.5 lib/`lib.es2015.symbol.wellknown.d.ts`/Math#
-//              ^^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Math#random().
+//         ^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Math#
+//         ^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Math.
+//         ^^^^ reference typescript 5.3.3 lib/`lib.es2015.core.d.ts`/Math#
+//         ^^^^ reference typescript 5.3.3 lib/`lib.es2015.symbol.wellknown.d.ts`/Math#
+//              ^^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Math#random().
   }
   export function returnStatement(): Configuration {
 //                ^^^^^^^^^^^^^^^ definition syntax 1.0.0 src/`infer-relationship.ts`/returnStatement().
@@ -68,11 +68,11 @@
 //                    documentation ```ts\n(property) '1': number\n```
       if (random() > Number.parseInt(i)) {
 //        ^^^^^^ reference syntax 1.0.0 src/`infer-relationship.ts`/random().
-//                   ^^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Number#
-//                   ^^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Number.
-//                   ^^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Number#
-//                   ^^^^^^ reference typescript 4.9.5 lib/`lib.es2020.number.d.ts`/Number#
-//                          ^^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.core.d.ts`/NumberConstructor#parseInt().
+//                   ^^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Number#
+//                   ^^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Number.
+//                   ^^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Number#
+//                   ^^^^^^ reference typescript 5.3.3 lib/`lib.es2020.number.d.ts`/Number#
+//                          ^^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.core.d.ts`/NumberConstructor#parseInt().
 //                                   ^ reference local 8
         return {
           property: 41,
@@ -117,7 +117,7 @@
 //                documentation ```ts\nfunction returnStatementInsideArgumentExpression(): Configuration[]\n```
 //                                                           ^^^^^^^^^^^^^ reference syntax 1.0.0 src/`infer-relationship.ts`/Configuration#
     return [1].map<Configuration>(number => {
-//             ^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Array#map().
+//             ^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Array#map().
 //                 ^^^^^^^^^^^^^ reference syntax 1.0.0 src/`infer-relationship.ts`/Configuration#
 //                                ^^^^^^ definition local 12
 //                                documentation ```ts\n(parameter) number: number\n```

--- a/snapshots/output/syntax/src/inheritance.ts
+++ b/snapshots/output/syntax/src/inheritance.ts
@@ -65,8 +65,8 @@
 //                   ^^^^^ definition syntax 1.0.0 src/`inheritance.ts`/Subclass#onLiteral().(param)
 //                   documentation ```ts\n(parameter) param: any\n```
       throw new Error('Method not implemented.' + param)
-//              ^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Error#
-//              ^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Error.
+//              ^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Error#
+//              ^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Error.
 //                                                ^^^^^ reference syntax 1.0.0 src/`inheritance.ts`/Subclass#onLiteral().(param)
     }
     property = 'property'
@@ -79,32 +79,32 @@
 //         relationship implementation reference scip-typescript npm syntax 1.0.0 src/`inheritance.ts`/IntermediateSuperclass#overrideMethod().
 //         relationship implementation reference scip-typescript npm syntax 1.0.0 src/`inheritance.ts`/Superclass#overrideMethod().
       throw new Error('Method not implemented.')
-//              ^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Error#
-//              ^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Error.
+//              ^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Error#
+//              ^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Error.
     }
     public intermediateOverrideMethod(): string {
 //         ^^^^^^^^^^^^^^^^^^^^^^^^^^ definition syntax 1.0.0 src/`inheritance.ts`/Subclass#intermediateOverrideMethod().
 //         documentation ```ts\n(method) intermediateOverrideMethod(): string\n```
 //         relationship implementation reference scip-typescript npm syntax 1.0.0 src/`inheritance.ts`/IntermediateSuperclass#intermediateOverrideMethod().
       throw new Error('Method not implemented.')
-//              ^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Error#
-//              ^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Error.
+//              ^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Error#
+//              ^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Error.
     }
     public interfaceMethod(): string {
 //         ^^^^^^^^^^^^^^^ definition syntax 1.0.0 src/`inheritance.ts`/Subclass#interfaceMethod().
 //         documentation ```ts\n(method) interfaceMethod(): string\n```
 //         relationship implementation reference scip-typescript npm syntax 1.0.0 src/`inheritance.ts`/Superinterface#interfaceMethod().
       throw new Error('Method not implemented.')
-//              ^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Error#
-//              ^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Error.
+//              ^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Error#
+//              ^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Error.
     }
     public intermediateInterfaceMethod(): string {
 //         ^^^^^^^^^^^^^^^^^^^^^^^^^^^ definition syntax 1.0.0 src/`inheritance.ts`/Subclass#intermediateInterfaceMethod().
 //         documentation ```ts\n(method) intermediateInterfaceMethod(): string\n```
 //         relationship implementation reference scip-typescript npm syntax 1.0.0 src/`inheritance.ts`/IntermediateSuperinterface#intermediateInterfaceMethod().
       throw new Error('Method not implemented.')
-//              ^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Error#
-//              ^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Error.
+//              ^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Error#
+//              ^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Error.
     }
   }
   export const objectLiteralImplementation: Superinterface = {
@@ -120,8 +120,8 @@
 //  documentation ```ts\n(property) interfaceMethod: () => string\n```
 //  relationship implementation reference scip-typescript npm syntax 1.0.0 src/`inheritance.ts`/Superinterface#interfaceMethod().
       throw new Error('Function not implemented.')
-//              ^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Error#
-//              ^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Error.
+//              ^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Error#
+//              ^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Error.
     },
   }
   export function consumesInterface(superInterface: Superinterface): void {}

--- a/snapshots/output/syntax/src/local.ts
+++ b/snapshots/output/syntax/src/local.ts
@@ -23,7 +23,7 @@
 //           documentation ```ts\nvar d: number\n```
 //                  ^ reference local 12
 //                      ^ reference local 8
-//                        ^^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/String#length.
+//                        ^^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/String#length.
 //                                ^ reference local 12
       c += d
 //    ^ reference local 8
@@ -31,12 +31,12 @@
       c2 += c.length
 //    ^^ reference local 9
 //          ^ reference local 8
-//            ^^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/String#length.
+//            ^^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/String#length.
     }
     return [c, c2].reduce((previousValue, currentValue, currentIndex) => {
 //          ^ reference local 8
 //             ^^ reference local 9
-//                 ^^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Array#reduce().
+//                 ^^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Array#reduce().
 //                         ^^^^^^^^^^^^^ definition local 16
 //                         documentation ```ts\n(parameter) previousValue: string\n```
 //                                        ^^^^^^^^^^^^ definition local 17

--- a/snapshots/output/syntax/src/string-literals.ts
+++ b/snapshots/output/syntax/src/string-literals.ts
@@ -19,11 +19,11 @@
   export type OmitInterface = Omit<SomeInterface, 'a' | 'b'>
 //            ^^^^^^^^^^^^^ definition syntax 1.0.0 src/`string-literals.ts`/OmitInterface#
 //            documentation ```ts\ntype OmitInterface\n```
-//                            ^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Omit#
+//                            ^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Omit#
 //                                 ^^^^^^^^^^^^^ reference syntax 1.0.0 src/`string-literals.ts`/SomeInterface#
   export type PickInterface = Pick<SomeInterface, 'b' | 'c'>
 //            ^^^^^^^^^^^^^ definition syntax 1.0.0 src/`string-literals.ts`/PickInterface#
 //            documentation ```ts\ntype PickInterface\n```
-//                            ^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Pick#
+//                            ^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Pick#
 //                                 ^^^^^^^^^^^^^ reference syntax 1.0.0 src/`string-literals.ts`/SomeInterface#
   

--- a/snapshots/output/syntax/src/structural-type.ts
+++ b/snapshots/output/syntax/src/structural-type.ts
@@ -3,34 +3,34 @@
 //documentation ```ts\nmodule "structural-type.ts"\n```
 //                ^^^ definition syntax 1.0.0 src/`structural-type.ts`/foo().
 //                documentation ```ts\nfunction foo(): Promise<{ member: number; }>\n```
-//                       ^^^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Promise#
-//                       ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.iterable.d.ts`/Promise#
-//                       ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.promise.d.ts`/Promise.
-//                       ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.symbol.wellknown.d.ts`/Promise#
-//                       ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2018.promise.d.ts`/Promise#
+//                       ^^^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Promise#
+//                       ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.iterable.d.ts`/Promise#
+//                       ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.promise.d.ts`/Promise.
+//                       ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.symbol.wellknown.d.ts`/Promise#
+//                       ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2018.promise.d.ts`/Promise#
 //                                 ^^^^^^ definition syntax 1.0.0 src/`structural-type.ts`/foo().Promise:typeLiteral0:member.
 //                                 documentation ```ts\n(property) member: number\n```
     return Promise.resolve({ member: 42 })
-//         ^^^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Promise#
-//         ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.iterable.d.ts`/Promise#
-//         ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.promise.d.ts`/Promise.
-//         ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.symbol.wellknown.d.ts`/Promise#
-//         ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2018.promise.d.ts`/Promise#
-//                 ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.promise.d.ts`/PromiseConstructor#resolve().
+//         ^^^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Promise#
+//         ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.iterable.d.ts`/Promise#
+//         ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.promise.d.ts`/Promise.
+//         ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.symbol.wellknown.d.ts`/Promise#
+//         ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2018.promise.d.ts`/Promise#
+//                 ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.promise.d.ts`/PromiseConstructor#resolve().
 //                           ^^^^^^ definition syntax 1.0.0 src/`structural-type.ts`/member0:
 //                           documentation ```ts\n(property) member: number\n```
   }
   export function bar(): Promise<number> {
 //                ^^^ definition syntax 1.0.0 src/`structural-type.ts`/bar().
 //                documentation ```ts\nfunction bar(): Promise<number>\n```
-//                       ^^^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Promise#
-//                       ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.iterable.d.ts`/Promise#
-//                       ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.promise.d.ts`/Promise.
-//                       ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.symbol.wellknown.d.ts`/Promise#
-//                       ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2018.promise.d.ts`/Promise#
+//                       ^^^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Promise#
+//                       ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.iterable.d.ts`/Promise#
+//                       ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.promise.d.ts`/Promise.
+//                       ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.symbol.wellknown.d.ts`/Promise#
+//                       ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2018.promise.d.ts`/Promise#
     return foo().then(x => x.member)
 //         ^^^ reference syntax 1.0.0 src/`structural-type.ts`/foo().
-//               ^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Promise#then().
+//               ^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Promise#then().
 //                    ^ definition local 4
 //                    documentation ```ts\n(parameter) x: { member: number; }\n```
 //                         ^ reference local 4
@@ -39,14 +39,14 @@
   export function bar2(): Promise<number> {
 //                ^^^^ definition syntax 1.0.0 src/`structural-type.ts`/bar2().
 //                documentation ```ts\nfunction bar2(): Promise<number>\n```
-//                        ^^^^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Promise#
-//                        ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.iterable.d.ts`/Promise#
-//                        ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.promise.d.ts`/Promise.
-//                        ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2015.symbol.wellknown.d.ts`/Promise#
-//                        ^^^^^^^ reference typescript 4.9.5 lib/`lib.es2018.promise.d.ts`/Promise#
+//                        ^^^^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Promise#
+//                        ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.iterable.d.ts`/Promise#
+//                        ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.promise.d.ts`/Promise.
+//                        ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2015.symbol.wellknown.d.ts`/Promise#
+//                        ^^^^^^^ reference typescript 5.3.3 lib/`lib.es2018.promise.d.ts`/Promise#
     return foo().then(({ member }) => member)
 //         ^^^ reference syntax 1.0.0 src/`structural-type.ts`/foo().
-//               ^^^^ reference typescript 4.9.5 lib/`lib.es5.d.ts`/Promise#then().
+//               ^^^^ reference typescript 5.3.3 lib/`lib.es5.d.ts`/Promise#then().
 //                       ^^^^^^ definition local 10
 //                       documentation ```ts\n(parameter) member: number\n```
 //                       ^^^^^^ reference syntax 1.0.0 src/`structural-type.ts`/foo().Promise:typeLiteral0:member.

--- a/snapshots/output/syntax/src/typings.ts
+++ b/snapshots/output/syntax/src/typings.ts
@@ -4,7 +4,7 @@
 //                ^^^^^^^ definition syntax 1.0.0 src/`typings.ts`/process().
 //                documentation ```ts\nfunction process(): Process\n```
     return window.process
-//         ^^^^^^ reference typescript 4.9.5 lib/`lib.dom.d.ts`/window.
+//         ^^^^^^ reference typescript 5.3.3 lib/`lib.dom.d.ts`/window.
 //                ^^^^^^^ reference @types/node 20.10.5 `globals.d.ts`/global/process.
 //                ^^^^^^^ reference @types/node 20.10.5 `process.d.ts`/`"process"`/global/process.
   }

--- a/src/FileIndexer.ts
+++ b/src/FileIndexer.ts
@@ -414,7 +414,7 @@ export class FileIndexer {
       if (props) {
         try {
           const tpe = this.checker.getTypeOfSymbolAtLocation(props, node)
-          const property = tpe.getProperty(node.name.text)
+          const property = tpe.getProperty(ts_inline.getTextOfJsxAttributeName(node.name))
           for (const decl of property?.declarations || []) {
             return this.scipSymbol(decl)
           }

--- a/src/FileIndexer.ts
+++ b/src/FileIndexer.ts
@@ -414,7 +414,9 @@ export class FileIndexer {
       if (props) {
         try {
           const tpe = this.checker.getTypeOfSymbolAtLocation(props, node)
-          const property = tpe.getProperty(ts_inline.getTextOfJsxAttributeName(node.name))
+          const property = tpe.getProperty(
+            ts_inline.getTextOfJsxAttributeName(node.name)
+          )
           for (const decl of property?.declarations || []) {
             return this.scipSymbol(decl)
           }

--- a/src/TypeScriptInternal.ts
+++ b/src/TypeScriptInternal.ts
@@ -58,3 +58,9 @@ export function isParameter(sym: ts.Symbol): boolean {
         : 'quit'
   )
 }
+
+// The corresponding function is marked @internal here:
+// https://sourcegraph.com/github.com/microsoft/TypeScript@fbcdb8cf4fbbbea0111a9adeb9d0d2983c088b7c/-/blob/src/compiler/utilities.ts?L10586-10589
+export function getTextOfJsxAttributeName(node: ts.JsxAttributeName): string {
+  return ts.isIdentifier(node) ? ts.idText(node) : `${ts.idText(node.namespace)}:${ts.idText(node.name)}`;
+}

--- a/src/TypeScriptInternal.ts
+++ b/src/TypeScriptInternal.ts
@@ -62,5 +62,7 @@ export function isParameter(sym: ts.Symbol): boolean {
 // The corresponding function is marked @internal here:
 // https://sourcegraph.com/github.com/microsoft/TypeScript@fbcdb8cf4fbbbea0111a9adeb9d0d2983c088b7c/-/blob/src/compiler/utilities.ts?L10586-10589
 export function getTextOfJsxAttributeName(node: ts.JsxAttributeName): string {
-  return ts.isIdentifier(node) ? ts.idText(node) : `${ts.idText(node.namespace)}:${ts.idText(node.name)}`;
+  return ts.isIdentifier(node)
+    ? ts.idText(node)
+    : `${ts.idText(node.namespace)}:${ts.idText(node.name)}`
 }

